### PR TITLE
feat: improve WriteApi performance

### PR DESCRIPTION
**Pull request by bnayae**
Some refactoring to the WriteApi which aimed to
having more performance & simplify the code.

This is a join effort of @WeKnow-network & @AviAvni.

We intend to provide better simplification of the code & better performance by converting the RX channel to TPL Dataflow or System.Threading.Channels.
We will provide this changes on separate Pull Request based on this PR.
We will maintain the RX pipeline as optional strategy until you'll decide to remove it.

### CODE REVIEW

**Comment by bednar** [comment]
Hi @bnayae,

thanks for your PR. Do you want continue with #96 or close it to favour #97? 

Regards

**Comment by bnayae** [comment]
this pull request depend on PointData Immutability #96.
Please check PointData Immutability #96 before this one.

**Comment by bednar** [comment]
Hi @bnayae, could you please rebase the PR to avoid conflicts?

**Comment by bnayae** [comment]
can you send the rebase command (and i will run it)?

File: Client/Writes/PointData.cs [hunk 1]
```diff
@@ -58,6 +58,19 @@ public static PointData Measurement(string measurementName)
...
+                            ImmutableSortedDictionary<string, string> tags,
+                            ImmutableSortedDictionary<string, object> fields)
+        {
+            _measurementName = measurementName;
+            Precision = precision;
+            _time = time;
+            _tags = tags;
+            _fields = fields;
+        }
+
```
Comment (bednar) [inline Client/Writes/PointData.cs:73]: This constructor is duplicated.

```
Writes/PointData.cs(61,17): error CS0111: Type 'PointData' already defines a member called '.ctor' with the same parameter types [/root/project/Client/Client.csproj]
    2 Warning(s)
    1 Error(s)
```

Please remove it.

File: Client/OrganizationsApi.cs [hunk 2]
```diff
@@ -141,7 +141,8 @@ public async Task<Organization> FindOrganizationByIdAsync(string orgId)
         /// <returns>List all organizations</returns>
         public async Task<List<Organization>> FindOrganizationsAsync()
         {
-            return await _service.GetOrgsAsync().ContinueWith(t => t.Result.Orgs);
+            var result = await _service.GetOrgsAsync();
+            return result.Orgs;
```
Comment (bednar) [inline Client/OrganizationsApi.cs:145]: Good point! We will fix it in whole client - see https://github.com/influxdata/influxdb-client-csharp/issues/108.

Could you please remove it from this PR.

File: Client/OrganizationsApi.cs [hunk 3]
```diff
@@ -139,9 +139,9 @@ public async Task<Organization> FindOrganizationByIdAsync(string orgId)
         /// List all organizations.
         /// </summary>
         /// <returns>List all organizations</returns>
-        public async Task<List<Organization>> FindOrganizationsAsync()
+        public Task<List<Organization>> FindOrganizationsAsync()
         {
-            return await _service.GetOrgsAsync().ContinueWith(t => t.Result.Orgs);
+            return _service.GetOrgsAsync().ContinueWith(t => t.Result.Orgs);
```
Comment (bednar) [inline Client/OrganizationsApi.cs:144]: Why are `async` and `await` removed?

```csharp
public async Task<List<Organization>> FindOrganizationsAsync()
{
    return await _service.GetOrgsAsync().ContinueWith(t => t.Result.Orgs);
}
```

Comment (bnayae) [reply Client/OrganizationsApi.cs:144]: I thought you asked for it
image

Comment (bednar) [reply Client/OrganizationsApi.cs:144]: Sorry for late. We've fixed the useless `ContinueWith` in #108. Could you rebase sources with master branch?

Comment (bnayae) [reply Client/OrganizationsApi.cs:144]: I don't sure how to rebase it, i already push it to the fork.
I can merge and remove this issue

**Pull request merged**